### DISCUSSION TITLE
Message body null, at specific whatsapp users

### DISCUSSION THREAD

**Discussion by juliherniyansah** [devlikeapro/waha#590] [General]
Hello

I've facing big problem when i want to read the body message, this is the JSON i got from webhook
```json
[
    {
        "id": {
            "server": "c.us",
            "user": "6288807419898",
            "_serialized": "6288807419898@c.us"
        },
        "name": "+62 888-0741-9898",
        "isGroup": false,
        "isReadOnly": false,
        "unreadCount": 0,
        "timestamp": 1728986772,
        "pinned": false,
        "muteExpiration": 0,
        "lastMessage": {
            "_data": {
                "id": {
                    "fromMe": false,
                    "remote": "6288807419898@c.us",
                    "id": "11E1DA11D8B8560561",
                    "_serialized": "false_6288807419898@c.us_11E1DA11D8B8560561"
                },
                "rowId": 1000000001,
                "viewed": false,
                "type": "biz_content_placeholder",
                "t": 1728986772,
                "from": {
                    "server": "c.us",
                    "user": "6288807419898",
                    "_serialized": "6288807419898@c.us"
                },
                "to": {
                    "server": "c.us",
                    "user": "6282196245286",
                    "_serialized": "6282196245286@c.us"
                },
                "ack": 3,
                "invis": true,
                "star": false,
                "kicNotified": false,
                "isFromTemplate": false,
                "pollOptions": [],
                "pollInvalidated": false,
                "pollVotesSnapshot": {
                    "pollVotes": []
                },
                "latestEditMsgKey": null,
                "latestEditSenderTimestampMs": null,
                "mentionedJidList": [],
                "groupMentions": [],
                "hsmTag": "AUTHENTICATION",
                "hsmCategory": null,
                "eventInvalidated": false,
                "isVcardOverMmsDocument": false,
                "isForwarded": false,
                "hasReaction": false,
                "viewMode": "VISIBLE",
                "privacyModeWhenSent": {
                    "actualActors": 2,
                    "hostStorage": 2,
                    "privacyModeTs": 1669611179
                },
                "productHeaderImageRejected": false,
                "lastPlaybackProgress": 0,
                "isDynamicReplyButtonsMsg": false,
                "isCarouselCard": false,
                "parentMsgId": null,
                "isMdHistoryMsg": false,
                "stickerSentTs": 0,
                "isAvatar": false,
                "lastUpdateFromServerTs": 0,
                "invokedBotWid": null,
                "bizBotType": null,
                "botResponseTargetId": null,
                "botPluginType": null,
                "botPluginReferenceIndex": null,
                "botPluginSearchProvider": null,
                "botPluginSearchUrl": null,
                "botPluginSearchQuery": null,
                "botPluginMaybeParent": false,
                "botReelPluginThumbnailCdnUrl": null,
                "botMsgBodyType": null,
                "requiresDirectConnection": false,
                "bizContentPlaceholderType": 0,
                "hostedBizEncStateMismatch": false,
                "senderOrRecipientAccountTypeHosted": false,
                "placeholderCreatedWhenAccountIsHosted": false,
                "links": []
            },
            "id": {
                "fromMe": false,
                "remote": "6288807419898@c.us",
                "id": "11E1DA11D8B8560561",
                "_serialized": "false_6288807419898@c.us_11E1DA11D8B8560561"
            },
            "ack": 3,
            "hasMedia": false,
            "body": "",
            "type": "biz_content_placeholder",
            "timestamp": 1728986772,
            "from": "6288807419898@c.us",
            "to": "xx@c.us",
            "deviceType": "web",
            "isForwarded": false,
            "forwardingScore": 0,
            "isStatus": false,
            "isStarred": false,
            "fromMe": false,
            "hasQuotedMsg": false,
            "hasReaction": false,
            "vCards": [],
            "mentionedIds": [],
            "groupMentions": [],
            "isGif": false,
            "links": []
        }
    },
    {
        "id": {
            "server": "c.us",
            "user": "xxx",
            "_serialized": "xx@c.us"
        },
        "name": "x",
        "isGroup": false,
        "unreadCount": 0,
        "pinned": false,
        "muteExpiration": 0
    }
]
```
I see field AUTHENTICATION on the return of the JSON, when i using screnshot it's there, message exist, please help me to fix it

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Hi! What is the message you see in the screenshot, could you share it?

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by juliherniyansah** [reply]
Sure, there any private channel that i can send u a screenshot, it's kind private for me

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by devlikepro** [reply]
waha@devlike.pro

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by juliherniyansah** [reply]
Hello, i had send you an email, and attach the screenshot, thanks before

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by devlikepro** [reply]
I see, it's OTP which WhatsApp doesn't send to the connected devices. No way to get it then in WEBJS or NOWEB, I don't thing they only hide it, they literlly don't send data to other devices, only to the primary one :( 
The message (the text) is on UI side, so you can some `biz_content_placeholder` or something
![image](https://github.com/user-attachments/assets/b966ab13-791f-4b86-b818-457f69cd5d93)


[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by devlikepro** [reply]
or check `invis`to show such messages on your UI if you're building one

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
> JSON i got from webhook

It looks like json response from one of API
Could you specify webhook event or API that you're using please?

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by juliherniyansah** [reply]
Yes, it's responses from /api/messages, cause i want to make sure the chatId is fit on the webhook, because my webhook showing this things too, i'm using message event to catch new message, and got body null for specific whatsapp users, for the rest of it, it's good, i can ready to read it, the body is there

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by devlikepro** [reply]
It might be "end to end encryption established" such message, need to chech that

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
